core/app/c/[communitySlug]/layout.tsx

```diff
@@ -26,7 +26,7 @@ export async function generateMetadata({
 }
 
 export default async function MainLayout({ children, params }: Props) {
-	const { user } = await getLoginData();
+	const { user } = await getPageLoginData();
```

Comment:
This was the important change, the others were just rote cleanup. But this makes sure logged in users get the proper login redirect rather than a `/settings` redirect.